### PR TITLE
kernel: mm: rename memory mapping functions from z_ to mm_memmap_

### PR DESCRIPTION
**Pull request by dcpleung**
This renames (mostly) the internal memory mapping functions from `z_` to have their own namespace `mm_memmap_` and `mm_vm_`. Also allows doxygen to start generating pages on them. See individual commits for details.

### CODE REVIEW

**Comment by npitre** [comment]
OK... but why? Some rationale would be nice.

Also, to the risk of bikeshedding a little, isn't MM_VM_ a bit generic?
At least the Z_ had the virtue of implying "Zephyr". Z_VM_ maybe?

**Comment by dcpleung** [comment]
One of the main reason is to get away from the very generic `Z_` namespace into its own `MM_` namespace, to signal these functions of related to memory management. Another main reason is to actually have doxygen generating documentation for these symbols, as anything beginning with `z_` or `Z_` are ignored.

As for the second level namespace, the `MM_VM_*` symbols are mostly symbols related to virtual memory, virtual addresses, their properties and the associated physical memory/address/properties. I can't think of a more concise naming. If you have other suggestions, please let me know.

**Comment by npitre** [comment]
Thanks for the background.

It is too bad that `z_` gets ignored by default. I'm not sure if
spreading into more namespaces for the system is a good idea. Seems to me 
that this gets in the way of applications.

Wouldn't `vm_*` instead of `mm_vm_*` be sufficient?

Also, wouldn't it make sense to have e.g.:

- `mm_memmap_phys_bare_map` --> `k_mem_map_phys_bare`

- `mm_memmap_phys_bare_unmap` --> `k_mem_unmap_phys_bare`

or even:

- `mm_memmap_phys_guard_unmap` --> `k_mm_unmap_phys_guard`

Because `mm_memmap_*_map` gets tripply redundant in the end.

**Comment by cfriedt** [comment]
Naming things is hard. The suggestions my @npitre are fine with me too though.

**Comment by npitre** [comment]
> One other thing, I remember that some guidelines about verb being the last in
> function name... probably need some massaging of those names.

IMHO: that depends.

If you have `k_mem_map` and `k_mem_unmap` then yes.

But when you have variations on the same theme I think it is far better not
to follow such rule. It is better to have `k_mem_map_phys` than
`k_mem_phys_map` and `k_mem_map_phys_bare` than `k_mem_phys_bare_map`.
In other words, variations should go last in the name so that when those
names are sorted then related functions get listed together rather than
being scattered around. Same argument goes for grepping.

**Comment by dcpleung** [comment]
Updated to use `k_mem_` as namespace. Except the mapping ones, others are mostly just doing a straight `z_` to `k_mem_` renaming.

**Comment by npitre** [comment]
Do you really want to expose `k_mem_map_phys_guard` as an official API?

I was under the impression that it would be better to encourage the use 
of `k_mem_phys_map` wrappers and actually change `k_mem_map_impl` to 
`z_mem_map_impl`. Those guard pages are still an internal implementation 
detail after all.

In fact, I was about to split physical and anon mem allocations appart 
as they are quite different already, and have guard page handling 
factored out as well to make the code cleaner.

**Comment by dcpleung** [comment]
> Do you really want to expose `k_mem_map_phys_guard` as an official API? I was under the impression that it would be better to encourage the use of `k_mem_phys_map` wrappers and actually change `k_mem_map_impl` to `z_mem_map_impl`. Those guard pages are still an internal implementation detail after all. In fact, I was about to split physical and anon mem allocations appart as they are quite different already, and have guard page handling factored out as well to make the code cleaner.

Both `k_mem_map_phys_guard/_bare` are still considered as kernel internal APIs (though not kernel private). Just like the old `z_phys_map` where you use it at your own risk. The naming is to make them more verbose so the caller knows whether guard pages are present, as moving `z_phys_map` out of `z_` namespace might create some confusions.

**Review by npitre — DISMISSED** [review]
Fair enough. In any case let's try to have this merged quickly as I have a bunch of demand paging patches yet to submit and this obviously creates conflicts all over the place.

**Review by andyross — DISMISSED** [review]
Not going to engage much in a naming discussion (except to say that let's please just do it **once** and leave it alone -- Zephyr's internal API commit history is extremely churn heavy as it is).   These seems fine to me.

Mostly these APIs exist for driver and soc integration, they aren't app code and don't need the kind of careful attention and love we give to our painstakingly designed and artisinally orthogonal IPC tools.  (Yes, that's a joke.)

File: arch/x86/core/efi.c [hunk 1]
```diff
@@ -33,8 +33,8 @@ void efi_init(struct efi_boot_arg *efi_arg)
 		return;
```
Comment (edersondisouza) [inline arch/x86/core/efi.c:33]: Nit: typo on commit message on "respectively". Maybe only worth fixing if there are more stuff to change.

**Review by edersondisouza — DISMISSED** [review]

**Comment by dcpleung** [comment]
Will fix the typos once the ACPICA PR is merged as I will need to modify the west manifest anyway.

**Comment by dcpleung** [comment]
Updated west manifest to point to actual commit and fixed the typos.

**Review by edersondisouza — APPROVED** [review]

**Review by npitre — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**